cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -460,23 +464,33 @@ Conflicts=umount.target
 [Mount]
 What=%[1]s
 Where=/sysroot
-%[2]s
-`, what, typ)
+%[2]s%[3]s
+`, what, typ, optionStr)
 	return content
 }
 
-func writeSysrootMountUnit(what, mntType string) error {
+func writeSysrootMountUnit(what, mntType string, opts fsOpts) (err error) {
+
 	// Writing the unit in this folder overrides
 	// /lib/systemd/system/sysroot-writable.mount - we will remove
 	// this unit eventually from the initramfs.
 	unitDir := dirs.SnapRuntimeServicesDirUnder(dirs.GlobalRootDir)
-	if err := os.MkdirAll(unitDir, 0755); err != nil {
+	if err = os.MkdirAll(unitDir, 0755); err != nil {
 		return err
 	}
 	unitFileName := "sysroot.mount"
 	unitPath := filepath.Join(unitDir, unitFileName)
 	what = dirs.StripRootDir(what)
-	unitContent := assembleSysrootMountUnitContent(what, mntType)
+
+	var options []string
+	if o, ok := opts.(*dmVerityOptions); ok && o != nil {
```

Comment:
can you explain why this is not just checking that opts != nil?

Reply:
+1, it could be other options not related to verity in the future.